### DISCUSSION TITLE
[classroom][actions] unable to fetch student's repo for auto-grading(actions)

### DISCUSSION THREAD

**Discussion by densuke-st** [community/community#17748] [GitHub Education]
The repository submitted by the student suddenly became unable to perform automatic evaluation.
This phenomenon is happening in the repository of all students. Why?

![image](https://user-images.githubusercontent.com/16230466/169158078-187a3878-fb2f-48ad-83fb-bf73649e8b47.png)
![image](https://user-images.githubusercontent.com/16230466/169158427-26b5e512-cc98-4750-8c9f-9098e49ee6ee.png)

**Comment by Sarabostani** [comment]
I'm facing the same issue. I thought it was related to [this incident](https://www.githubstatus.com/incidents/bq3s2sp6sfch) but the problem still hasn't been resolved.

**Reply by JessRudder** [reply]
Hello! We believe we've pushed a fix that should get the student repos back into a good state. Could you please let me know if you're still seeing the same issue?

**Reply by hanniemirac** [reply]
Hi, I'm almost a year late however, the issue still persists for me. Any suggestions?

**Comment by bharlowRHC** [comment]
I'm facing the same issue also with my students. It seemed fine prior to the reported incident, but after I get the same result as the example screenshot in this post.

**Reply by densuke-st** [reply]
How to escalate this?

**Reply by bharlowRHC** [reply]
@elisemoe How can we escalate this? 

Monday prior to the [incident](https://www.githubstatus.com/incidents/bq3s2sp6sfch), I helped students without any difficulty and saw how it successfully graded the assignments. Then on Tuesday after the incident, for the same assignments, I get the same error message as @densuke-st reported saying it cannot find the repositories.

I'm happy to DM example assignments links if it helps.

<img width="920" alt="github-autograde-error-128" src="https://user-images.githubusercontent.com/37458144/169200794-d7bd877f-1576-460e-8f35-cc4d87ad2795.png">

**Reply by densuke-st** [reply]
!!! very Thanks!

**Reply by mstrommer** [reply]
I changed my test repo to public and then it worked. So the issue arises from private student repositories. But students can not change the visibility of their projects...

**Reply by mstrommer** [reply]
Changing repo visibility to public within the admin interface, after repos have been created, reflects immediately within student repos. Autograding is triggered and performs fine. This is of course just a workaround.

**Reply by JessRudder** [reply]
Hello! We believe we've pushed a fix that should get the student repos back into a good state. Could you please let me know if you're still seeing the same issue?

**Reply by mstrommer** [reply]
With existing repos the error still persists. With newly accepted assignments it works. It would be cool if the problem gets also solved for existing repos.

**Comment by bharlowRHC** [comment]
@mstrommer Yes, you are absolutely right. I fired up my practice repo we use for students to practice git commands and it worked when public, but as soon as I made it private, it failed. Unfortunately, it's not a workaround I can use.

@elisemoe Below are screenshots from my practice repo, this GitHub Classroom assignment is created private and fails the Auto Grading Unit Tests.

![Screen Shot 2022-05-19 at 11 35 57 AM](https://user-images.githubusercontent.com/37458144/169376657-fe1d2d60-ddb5-473f-ad12-d32abf79676d.png)

However, after switching it to public and making no other changes before doing a re-run of the same job, it works fine.

![Screen Shot 2022-05-19 at 11 34 19 AM](https://user-images.githubusercontent.com/37458144/169376766-6a61a902-4be6-4914-a63a-0ad4db69ea01.png)

**Reply by densuke-st** [reply]
Thanks!

I also changed to a public repository and it worked fine.
However, when I revert to the private repository again, it fails as before.
Since it is a student submission, I would like to return it to a private repository. What should I do?

**Reply by bharlowRHC** [reply]
You can always revert it back to private if the assignment is done. It will not automatically re-run the job by changing it.

I am going to use the [GitHub Classroom Assistant](https://classroom.github.com/assistant) tool to suck down all the repo's then manually run my unit test. I have too many assignments to go in an flip them one at a time (35 students x 2 modules x 2 assignments per module) and I am almost done with the semester.

**Reply by Sarabostani** [reply]
Does this happen when new assignments are created as well or just the existing ones?

**Reply by bharlowRHC** [reply]
Looking at last weeks assignment, the students who started prior to the incident have been able to get the auto grade to pass or get a proper report on why the grading did not pass. The students who started after the incident, have received the repo not found error. 

My practice repo was created this morning and it failed when set to private.

I will try creating a brand new assignment in addition to using an existing one.

**Reply by mstrommer** [reply]
It happens with all newly accepted assignments. Old ones just work fine.

**Reply by bharlowRHC** [reply]
I just created a new template repo with a new assignment then accepted it, the issue still happens.

<img width="939" alt="bug-hunt" src="https://user-images.githubusercontent.com/37458144/169390710-287b5a05-13bd-44a8-bd7a-97b9f89039e8.png">

**Reply by JessRudder** [reply]
Hello! We believe we've pushed a fix that should get the student repos back into a good state. Could you please let me know if you're still seeing the same issue?

**Reply by bharlowRHC** [reply]
Still have the same issue after a rerun and a new commit.

<img width="845" alt="bug-hunt-2" src="https://user-images.githubusercontent.com/37458144/169435021-a4a6a2df-9715-4e10-9101-00c57d44a8e0.png">

**Reply by octosteve** [reply]
@bharlowRHC Can you add me to the student's repository? I'd love to take a look.

**Comment by kootsoop** [comment]
Same problem. Would be nice to get a fix!

I can confirm that making the student repos public "fixes" the problem, but that's not really the way I'd like to fix it.

**Reply by JessRudder** [reply]
Hello! We believe we've pushed a fix that should get the student repos back into a good state. Could you please let me know if you're still seeing the same issue?

**Comment by densuke-st** [comment]
workaround?

1. download students repos using [GitHub Classroom Assistant](https://classroom.github.com/assistant) tool (Thanks! @bharlowRHC )
2. cd /tmp/XXXXXX
3.  `for i in *; do cd $i; gh repo edit --visibility public; cd -; done`
4. open student repos in browser, and go actions -> latest workflow -> re-run jobs # very tired...
5.  `for i in *; do cd $i; gh repo edit --visibility private; cd -; done`
6. get result in assignment.

**Reply by densuke-st** [reply]
if passed deadline, please move deadline.

**Reply by JessRudder** [reply]
Hello! We believe we've pushed a fix that should get the student repos back into a good state. Could you please let me know if you're still seeing the same issue?

**Reply by Sarabostani** [reply]
@JessRudder I ran the failed action again on a student's repo but still see the same error.

**Reply by densuke-st** [reply]
@JessRudder  Thanks!

But... Still failed.

![image](https://user-images.githubusercontent.com/16230466/169428141-369ee750-5588-4cb5-adce-906177e078ca.png)

**Reply by kootsoop** [reply]
I concur: the pushed fix does not help.

**Reply by octosteve** [reply]
Can you add me to the the affected students' repositories? @kootsoop @Sarabostani. I'd love to take a look.

**Comment by Sarabostani** [comment]
@JessRudder one of my students found out that by modifying the permission block of the workflow the autograder was able to pull the repository and run
![image](https://user-images.githubusercontent.com/4537598/169569692-ba0e2a2b-0f2a-49a0-9577-06d67e22b323.png)

This is not a solution or a feasible workaround but I thought it could give some insight to what might have changed.

**Reply by octosteve** [reply]
I'm looking at your Autograding workflow... it doesn't seem to be the one generated from Classroom. Is that right? The `env` settings aren't something we generate. If you supplied your own workflow, we can't automatically fix it since it would override your settings. Did you generate this workflow yourself @Sarabostani?

**Reply by Sarabostani** [reply]
No I didn't supply my own workflow. It gets generated when I create a new assignment. The student modified the workflow however to make the autograder work

**Reply by octosteve** [reply]
Thanks for getting back to me. Could you add me to the student's repository? I'd love to to take a look.

**Reply by Sarabostani** [reply]
I added you. I just compared the workflow with the previous assignments. The new ones use `checkout@v3` and the `env` and `permissions` config keys have been added. Below is the workflow from an older assignment.

         name: GitHub Classroom Workflow
 
     on: [push]
 
     jobs:
       build:
         name: Autograding
         runs-on: ubuntu-latest
         steps:
           - uses: actions/checkout@v2
           - uses: education/autograding@v1

**Reply by Sarabostani** [reply]
@octosteve I added you to another repo as well. I just commented out the `permissions` config and the workflow was able to pull the repo.

**Reply by octosteve** [reply]
Great! I just joined. The `v3` checkout is a bit of a mystery... We don't have any codepaths that would set that to v3.

**Reply by octosteve** [reply]
Looking at the history, it looks like something the student changed.

**Reply by Sarabostani** [reply]
Please ignore that part, that might be something that the student might have tried out. I looked at the history of the commit and there was no `v3` for the checkout process :)

**Reply by octosteve** [reply]
The _new_ permissions should be 
```
permissions:
  checks: write
  actions: read
  contents: read
```

If you add these, your tests should work as expected. This is the fix we should have pushed out to your student's repo. It looks like it didn't, based on the git history. Let me take a look and see if our filters were right when we targeted repos.

**Reply by Sarabostani** [reply]
Yes! That did the trick. Thank you for looking into this. I'll update the students' repository manually for the existing assignment.

**Reply by bharlowRHC** [reply]
@Sarabostani Just update the Classroom assignment. I modified one of the names of the Unit Tests then saved the change. That was enough to update all of the students repositories at once rather than do a one at a time update and since it was a commit it also reran the action. I just watched my numbers go from 10 passing to 15 passing by doing that.

**Reply by Sarabostani** [reply]
Thank you! That worked :)

**Reply by densuke-st** [reply]
Thank you! Our repo worked!
All students repo append 'contents' permission...

**Comment by bharlowRHC** [comment]
For the GitHub Staff who are working the problem, thank you! Here is my bug hunt assignment: https://classroom.github.com/a/msftiH0e

This is setup exactly as my students would have it with the exception I have given you admin access. Mine still fails and I have not modified any of the settings.

I will delete the assignment once it is resolved.

**Reply by octosteve** [reply]
![image](https://user-images.githubusercontent.com/325485/169575213-64fdd942-2524-4550-87ce-82efb15b9bcc.png)
It looks like it's failing, but for the right reasons... I haven't passed the tests.

Can you add me to your repository that's failing?

**Reply by bharlowRHC** [reply]
Done, you have full admin access to my repo that was failing, but now has the green checkmark that it passed!
